{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8891e3f7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting unstructured==0.11.6\n",
      "  Using cached unstructured-0.11.6-py3-none-any.whl.metadata (26 kB)\n",
      "Collecting chardet (from unstructured==0.11.6)\n",
      "  Using cached chardet-5.2.0-py3-none-any.whl.metadata (3.4 kB)\n",
      "Collecting filetype (from unstructured==0.11.6)\n",
      "  Downloading filetype-1.2.0-py2.py3-none-any.whl.metadata (6.5 kB)\n",
      "Collecting python-magic (from unstructured==0.11.6)\n",
      "  Downloading python_magic-0.4.27-py2.py3-none-any.whl.metadata (5.8 kB)\n",
      "Collecting lxml (from unstructured==0.11.6)\n",
      "  Downloading lxml-5.1.0-cp38-cp38-win_amd64.whl.metadata (3.6 kB)\n",
      "Requirement already satisfied: nltk in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from unstructured==0.11.6) (3.8.1)\n",
      "Collecting tabulate (from unstructured==0.11.6)\n",
      "  Downloading tabulate-0.9.0-py3-none-any.whl.metadata (34 kB)\n",
      "Requirement already satisfied: requests in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from unstructured==0.11.6) (2.31.0)\n",
      "Requirement already satisfied: beautifulsoup4 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from unstructured==0.11.6) (4.12.3)\n",
      "Collecting emoji (from unstructured==0.11.6)\n",
      "  Downloading emoji-2.10.1-py2.py3-none-any.whl.metadata (5.3 kB)\n",
      "Requirement already satisfied: dataclasses-json in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from unstructured==0.11.6) (0.6.4)\n",
      "Collecting python-iso639 (from unstructured==0.11.6)\n",
      "  Downloading python_iso639-2024.2.7-py3-none-any.whl.metadata (13 kB)\n",
      "Collecting langdetect (from unstructured==0.11.6)\n",
      "  Using cached langdetect-1.0.9-py3-none-any.whl\n",
      "Requirement already satisfied: numpy in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from unstructured==0.11.6) (1.24.4)\n",
      "Collecting rapidfuzz (from unstructured==0.11.6)\n",
      "  Downloading rapidfuzz-3.6.2-cp38-cp38-win_amd64.whl.metadata (11 kB)\n",
      "Collecting backoff (from unstructured==0.11.6)\n",
      "  Downloading backoff-2.2.1-py3-none-any.whl.metadata (14 kB)\n",
      "Requirement already satisfied: typing-extensions in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from unstructured==0.11.6) (4.10.0)\n",
      "Collecting unstructured-client (from unstructured==0.11.6)\n",
      "  Downloading unstructured_client-0.22.0-py3-none-any.whl.metadata (7.3 kB)\n",
      "Collecting wrapt (from unstructured==0.11.6)\n",
      "  Using cached wrapt-1.16.0-cp38-cp38-win_amd64.whl.metadata (6.8 kB)\n",
      "Requirement already satisfied: soupsieve>1.2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from beautifulsoup4->unstructured==0.11.6) (2.5)\n",
      "Requirement already satisfied: marshmallow<4.0.0,>=3.18.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from dataclasses-json->unstructured==0.11.6) (3.21.1)\n",
      "Requirement already satisfied: typing-inspect<1,>=0.4.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from dataclasses-json->unstructured==0.11.6) (0.9.0)\n",
      "Requirement already satisfied: six in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from langdetect->unstructured==0.11.6) (1.16.0)\n",
      "Requirement already satisfied: click in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from nltk->unstructured==0.11.6) (8.1.7)\n",
      "Requirement already satisfied: joblib in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from nltk->unstructured==0.11.6) (1.3.2)\n",
      "Requirement already satisfied: regex>=2021.8.3 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from nltk->unstructured==0.11.6) (2023.12.25)\n",
      "Requirement already satisfied: tqdm in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from nltk->unstructured==0.11.6) (4.66.2)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->unstructured==0.11.6) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->unstructured==0.11.6) (3.6)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->unstructured==0.11.6) (2.2.1)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->unstructured==0.11.6) (2024.2.2)\n",
      "Collecting deepdiff>=6.0 (from unstructured-client->unstructured==0.11.6)\n",
      "  Downloading deepdiff-6.7.1-py3-none-any.whl.metadata (6.1 kB)\n",
      "Collecting jsonpath-python>=1.0.6 (from unstructured-client->unstructured==0.11.6)\n",
      "  Downloading jsonpath_python-1.0.6-py3-none-any.whl.metadata (12 kB)\n",
      "Requirement already satisfied: mypy-extensions>=1.0.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from unstructured-client->unstructured==0.11.6) (1.0.0)\n",
      "Requirement already satisfied: packaging>=23.1 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from unstructured-client->unstructured==0.11.6) (23.2)\n",
      "Collecting pypdf>=4.0 (from unstructured-client->unstructured==0.11.6)\n",
      "  Downloading pypdf-4.1.0-py3-none-any.whl.metadata (7.4 kB)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from unstructured-client->unstructured==0.11.6) (2.9.0.post0)\n",
      "Collecting ordered-set<4.2.0,>=4.0.2 (from deepdiff>=6.0->unstructured-client->unstructured==0.11.6)\n",
      "  Downloading ordered_set-4.1.0-py3-none-any.whl.metadata (5.3 kB)\n",
      "Requirement already satisfied: colorama in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from click->nltk->unstructured==0.11.6) (0.4.6)\n",
      "Using cached unstructured-0.11.6-py3-none-any.whl (1.8 MB)\n",
      "Using cached backoff-2.2.1-py3-none-any.whl (15 kB)\n",
      "Using cached chardet-5.2.0-py3-none-any.whl (199 kB)\n",
      "Downloading emoji-2.10.1-py2.py3-none-any.whl (421 kB)\n",
      "   ---------------------------------------- 0.0/421.5 kB ? eta -:--:--\n",
      "   --------------------------------------  419.8/421.5 kB 13.2 MB/s eta 0:00:01\n",
      "   ---------------------------------------- 421.5/421.5 kB 6.5 MB/s eta 0:00:00\n",
      "Using cached filetype-1.2.0-py2.py3-none-any.whl (19 kB)\n",
      "Downloading lxml-5.1.0-cp38-cp38-win_amd64.whl (3.9 MB)\n",
      "   ---------------------------------------- 0.0/3.9 MB ? eta -:--:--\n",
      "   ------ --------------------------------- 0.6/3.9 MB 12.4 MB/s eta 0:00:01\n",
      "   --------- ------------------------------ 0.9/3.9 MB 11.7 MB/s eta 0:00:01\n",
      "   -------------- ------------------------- 1.4/3.9 MB 9.9 MB/s eta 0:00:01\n",
      "   -------------------- ------------------- 2.0/3.9 MB 10.6 MB/s eta 0:00:01\n",
      "   --------------------------- ------------ 2.7/3.9 MB 11.5 MB/s eta 0:00:01\n",
      "   -------------------------------- ------- 3.2/3.9 MB 11.4 MB/s eta 0:00:01\n",
      "   -------------------------------------- - 3.8/3.9 MB 11.5 MB/s eta 0:00:01\n",
      "   ---------------------------------------- 3.9/3.9 MB 10.4 MB/s eta 0:00:00\n",
      "Downloading python_iso639-2024.2.7-py3-none-any.whl (274 kB)\n",
      "   ---------------------------------------- 0.0/274.7 kB ? eta -:--:--\n",
      "   ---------------------------------------- 274.7/274.7 kB 8.3 MB/s eta 0:00:00\n",
      "Using cached python_magic-0.4.27-py2.py3-none-any.whl (13 kB)\n",
      "Downloading rapidfuzz-3.6.2-cp38-cp38-win_amd64.whl (1.6 MB)\n",
      "   ---------------------------------------- 0.0/1.6 MB ? eta -:--:--\n",
      "   ----------------- ---------------------- 0.7/1.6 MB 22.5 MB/s eta 0:00:01\n",
      "   ------------------------------ --------- 1.3/1.6 MB 16.1 MB/s eta 0:00:01\n",
      "   ---------------------------------------  1.6/1.6 MB 14.9 MB/s eta 0:00:01\n",
      "   ---------------------------------------- 1.6/1.6 MB 11.7 MB/s eta 0:00:00\n",
      "Using cached tabulate-0.9.0-py3-none-any.whl (35 kB)\n",
      "Downloading unstructured_client-0.22.0-py3-none-any.whl (28 kB)\n",
      "Using cached wrapt-1.16.0-cp38-cp38-win_amd64.whl (37 kB)\n",
      "Downloading deepdiff-6.7.1-py3-none-any.whl (76 kB)\n",
      "   ---------------------------------------- 0.0/76.6 kB ? eta -:--:--\n",
      "   ---------------------------------------- 76.6/76.6 kB 2.1 MB/s eta 0:00:00\n",
      "Using cached jsonpath_python-1.0.6-py3-none-any.whl (7.6 kB)\n",
      "Downloading pypdf-4.1.0-py3-none-any.whl (286 kB)\n",
      "   ---------------------------------------- 0.0/286.1 kB ? eta -:--:--\n",
      "   ---------------------------------------- 286.1/286.1 kB 8.6 MB/s eta 0:00:00\n",
      "Downloading ordered_set-4.1.0-py3-none-any.whl (7.6 kB)\n",
      "Installing collected packages: filetype, wrapt, tabulate, rapidfuzz, python-magic, python-iso639, pypdf, ordered-set, lxml, langdetect, jsonpath-python, emoji, chardet, backoff, deepdiff, unstructured-client, unstructured\n",
      "  Attempting uninstall: pypdf\n",
      "    Found existing installation: pypdf 3.17.4\n",
      "    Uninstalling pypdf-3.17.4:\n",
      "      Successfully uninstalled pypdf-3.17.4\n",
      "Successfully installed backoff-2.2.1 chardet-5.2.0 deepdiff-6.7.1 emoji-2.10.1 filetype-1.2.0 jsonpath-python-1.0.6 langdetect-1.0.9 lxml-5.1.0 ordered-set-4.1.0 pypdf-4.1.0 python-iso639-2024.2.7 python-magic-0.4.27 rapidfuzz-3.6.2 tabulate-0.9.0 unstructured-0.11.6 unstructured-client-0.22.0 wrapt-1.16.0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "    WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n"
     ]
    }
   ],
   "source": [
    "!pip install unstructured==0.11.6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "da9f9c50",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: sentence-transformers==2.2.2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (2.2.2)\n",
      "Requirement already satisfied: transformers<5.0.0,>=4.6.0 in c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages (from sentence-transformers==2.2.2) (4.12.5)\n",
      "Requirement already satisfied: tqdm in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (4.66.2)\n",
      "Requirement already satisfied: torch>=1.6.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (2.2.1)\n",
      "Requirement already satisfied: torchvision in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (0.17.1)\n",
      "Requirement already satisfied: numpy in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (1.24.4)\n",
      "Requirement already satisfied: scikit-learn in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (1.3.2)\n",
      "Requirement already satisfied: scipy in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (1.10.1)\n",
      "Requirement already satisfied: nltk in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (3.8.1)\n",
      "Requirement already satisfied: sentencepiece in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (0.2.0)\n",
      "Requirement already satisfied: huggingface-hub>=0.4.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (0.19.4)\n",
      "Requirement already satisfied: filelock in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (3.13.1)\n",
      "Requirement already satisfied: fsspec>=2023.5.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (2024.3.0)\n",
      "Requirement already satisfied: requests in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (2.31.0)\n",
      "Requirement already satisfied: pyyaml>=5.1 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (6.0.1)\n",
      "Requirement already satisfied: typing-extensions>=3.7.4.3 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (4.10.0)\n",
      "Requirement already satisfied: packaging>=20.9 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (23.2)\n",
      "Requirement already satisfied: sympy in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from torch>=1.6.0->sentence-transformers==2.2.2) (1.12)\n",
      "Requirement already satisfied: networkx in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from torch>=1.6.0->sentence-transformers==2.2.2) (3.1)\n",
      "Requirement already satisfied: jinja2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from torch>=1.6.0->sentence-transformers==2.2.2) (3.1.3)\n",
      "Requirement already satisfied: colorama in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from tqdm->sentence-transformers==2.2.2) (0.4.6)\n",
      "Requirement already satisfied: regex!=2019.12.17 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from transformers<5.0.0,>=4.6.0->sentence-transformers==2.2.2) (2023.12.25)\n",
      "Requirement already satisfied: sacremoses in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from transformers<5.0.0,>=4.6.0->sentence-transformers==2.2.2) (0.1.1)\n",
      "Requirement already satisfied: tokenizers<0.11,>=0.10.1 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from transformers<5.0.0,>=4.6.0->sentence-transformers==2.2.2) (0.10.3)\n",
      "Requirement already satisfied: click in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from nltk->sentence-transformers==2.2.2) (8.1.7)\n",
      "Requirement already satisfied: joblib in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from nltk->sentence-transformers==2.2.2) (1.3.2)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from scikit-learn->sentence-transformers==2.2.2) (3.3.0)\n",
      "Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from torchvision->sentence-transformers==2.2.2) (10.2.0)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from jinja2->torch>=1.6.0->sentence-transformers==2.2.2) (2.1.5)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (3.6)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (2.2.1)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (2024.2.2)\n",
      "Requirement already satisfied: mpmath>=0.19 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sympy->torch>=1.6.0->sentence-transformers==2.2.2) (1.3.0)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n"
     ]
    }
   ],
   "source": [
    "!pip install sentence-transformers==2.2.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7fa3dc87",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting chromadb==0.4.21\n",
      "  Using cached chromadb-0.4.21-py3-none-any.whl.metadata (7.3 kB)\n",
      "Requirement already satisfied: requests>=2.28 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from chromadb==0.4.21) (2.31.0)\n",
      "Requirement already satisfied: pydantic>=1.9 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from chromadb==0.4.21) (2.6.4)\n",
      "Collecting chroma-hnswlib==0.7.3 (from chromadb==0.4.21)\n",
      "  Using cached chroma_hnswlib-0.7.3-cp38-cp38-win_amd64.whl.metadata (262 bytes)\n",
      "Collecting fastapi>=0.95.2 (from chromadb==0.4.21)\n",
      "  Downloading fastapi-0.110.0-py3-none-any.whl.metadata (25 kB)\n",
      "Collecting uvicorn>=0.18.3 (from uvicorn[standard]>=0.18.3->chromadb==0.4.21)\n",
      "  Downloading uvicorn-0.28.0-py3-none-any.whl.metadata (6.3 kB)\n",
      "Requirement already satisfied: numpy>=1.22.5 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from chromadb==0.4.21) (1.24.4)\n",
      "Collecting posthog>=2.4.0 (from chromadb==0.4.21)\n",
      "  Downloading posthog-3.5.0-py2.py3-none-any.whl.metadata (2.0 kB)\n",
      "Requirement already satisfied: typing-extensions>=4.5.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from chromadb==0.4.21) (4.10.0)\n",
      "Collecting pulsar-client>=3.1.0 (from chromadb==0.4.21)\n",
      "  Downloading pulsar_client-3.4.0-cp38-cp38-win_amd64.whl.metadata (1.0 kB)\n",
      "Collecting onnxruntime>=1.14.1 (from chromadb==0.4.21)\n",
      "  Downloading onnxruntime-1.17.1-cp38-cp38-win_amd64.whl.metadata (4.4 kB)\n",
      "Collecting opentelemetry-api>=1.2.0 (from chromadb==0.4.21)\n",
      "  Downloading opentelemetry_api-1.23.0-py3-none-any.whl.metadata (1.4 kB)\n",
      "Collecting opentelemetry-exporter-otlp-proto-grpc>=1.2.0 (from chromadb==0.4.21)\n",
      "  Downloading opentelemetry_exporter_otlp_proto_grpc-1.23.0-py3-none-any.whl.metadata (2.2 kB)\n",
      "Collecting opentelemetry-instrumentation-fastapi>=0.41b0 (from chromadb==0.4.21)\n",
      "  Downloading opentelemetry_instrumentation_fastapi-0.44b0-py3-none-any.whl.metadata (2.3 kB)\n",
      "Collecting opentelemetry-sdk>=1.2.0 (from chromadb==0.4.21)\n",
      "  Downloading opentelemetry_sdk-1.23.0-py3-none-any.whl.metadata (1.4 kB)\n",
      "Collecting tokenizers>=0.13.2 (from chromadb==0.4.21)\n",
      "  Using cached tokenizers-0.15.2-cp38-none-win_amd64.whl.metadata (6.8 kB)\n",
      "Collecting pypika>=0.48.9 (from chromadb==0.4.21)\n",
      "  Using cached PyPika-0.48.9-py2.py3-none-any.whl\n",
      "Requirement already satisfied: tqdm>=4.65.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from chromadb==0.4.21) (4.66.2)\n",
      "Collecting overrides>=7.3.1 (from chromadb==0.4.21)\n",
      "  Downloading overrides-7.7.0-py3-none-any.whl.metadata (5.8 kB)\n",
      "Requirement already satisfied: importlib-resources in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from chromadb==0.4.21) (6.3.1)\n",
      "Collecting grpcio>=1.58.0 (from chromadb==0.4.21)\n",
      "  Downloading grpcio-1.62.1-cp38-cp38-win_amd64.whl.metadata (4.2 kB)\n",
      "Collecting bcrypt>=4.0.1 (from chromadb==0.4.21)\n",
      "  Using cached bcrypt-4.1.2-cp37-abi3-win_amd64.whl.metadata (9.8 kB)\n",
      "Collecting typer>=0.9.0 (from chromadb==0.4.21)\n",
      "  Downloading typer-0.9.0-py3-none-any.whl.metadata (14 kB)\n",
      "Collecting kubernetes>=28.1.0 (from chromadb==0.4.21)\n",
      "  Downloading kubernetes-29.0.0-py2.py3-none-any.whl.metadata (1.5 kB)\n",
      "Requirement already satisfied: tenacity>=8.2.3 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from chromadb==0.4.21) (8.2.3)\n",
      "Requirement already satisfied: PyYAML>=6.0.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from chromadb==0.4.21) (6.0.1)\n",
      "Collecting mmh3>=4.0.1 (from chromadb==0.4.21)\n",
      "  Downloading mmh3-4.1.0-cp38-cp38-win_amd64.whl.metadata (13 kB)\n",
      "Collecting graphlib-backport>=1.0.3 (from chromadb==0.4.21)\n",
      "  Downloading graphlib_backport-1.1.0-py3-none-any.whl.metadata (4.4 kB)\n",
      "Collecting starlette<0.37.0,>=0.36.3 (from fastapi>=0.95.2->chromadb==0.4.21)\n",
      "  Downloading starlette-0.36.3-py3-none-any.whl.metadata (5.9 kB)\n",
      "Requirement already satisfied: certifi>=14.05.14 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from kubernetes>=28.1.0->chromadb==0.4.21) (2024.2.2)\n",
      "Requirement already satisfied: six>=1.9.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from kubernetes>=28.1.0->chromadb==0.4.21) (1.16.0)\n",
      "Requirement already satisfied: python-dateutil>=2.5.3 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from kubernetes>=28.1.0->chromadb==0.4.21) (2.9.0.post0)\n",
      "Collecting google-auth>=1.0.1 (from kubernetes>=28.1.0->chromadb==0.4.21)\n",
      "  Downloading google_auth-2.28.2-py2.py3-none-any.whl.metadata (4.7 kB)\n",
      "Collecting websocket-client!=0.40.0,!=0.41.*,!=0.42.*,>=0.32.0 (from kubernetes>=28.1.0->chromadb==0.4.21)\n",
      "  Using cached websocket_client-1.7.0-py3-none-any.whl.metadata (7.9 kB)\n",
      "Collecting requests-oauthlib (from kubernetes>=28.1.0->chromadb==0.4.21)\n",
      "  Downloading requests_oauthlib-1.4.0-py2.py3-none-any.whl.metadata (11 kB)\n",
      "Collecting oauthlib>=3.2.2 (from kubernetes>=28.1.0->chromadb==0.4.21)\n",
      "  Downloading oauthlib-3.2.2-py3-none-any.whl.metadata (7.5 kB)\n",
      "Requirement already satisfied: urllib3>=1.24.2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from kubernetes>=28.1.0->chromadb==0.4.21) (2.2.1)\n",
      "Collecting coloredlogs (from onnxruntime>=1.14.1->chromadb==0.4.21)\n",
      "  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl.metadata (12 kB)\n",
      "Collecting flatbuffers (from onnxruntime>=1.14.1->chromadb==0.4.21)\n",
      "  Downloading flatbuffers-24.3.7-py2.py3-none-any.whl.metadata (849 bytes)\n",
      "Requirement already satisfied: packaging in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from onnxruntime>=1.14.1->chromadb==0.4.21) (23.2)\n",
      "Requirement already satisfied: protobuf in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from onnxruntime>=1.14.1->chromadb==0.4.21) (4.25.3)\n",
      "Requirement already satisfied: sympy in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from onnxruntime>=1.14.1->chromadb==0.4.21) (1.12)\n",
      "Collecting deprecated>=1.2.6 (from opentelemetry-api>=1.2.0->chromadb==0.4.21)\n",
      "  Using cached Deprecated-1.2.14-py2.py3-none-any.whl.metadata (5.4 kB)\n",
      "Requirement already satisfied: importlib-metadata<7.0,>=6.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from opentelemetry-api>=1.2.0->chromadb==0.4.21) (6.11.0)\n",
      "Collecting googleapis-common-protos~=1.52 (from opentelemetry-exporter-otlp-proto-grpc>=1.2.0->chromadb==0.4.21)\n",
      "  Downloading googleapis_common_protos-1.63.0-py2.py3-none-any.whl.metadata (1.5 kB)\n",
      "Collecting opentelemetry-exporter-otlp-proto-common==1.23.0 (from opentelemetry-exporter-otlp-proto-grpc>=1.2.0->chromadb==0.4.21)\n",
      "  Downloading opentelemetry_exporter_otlp_proto_common-1.23.0-py3-none-any.whl.metadata (1.7 kB)\n",
      "Collecting opentelemetry-proto==1.23.0 (from opentelemetry-exporter-otlp-proto-grpc>=1.2.0->chromadb==0.4.21)\n",
      "  Downloading opentelemetry_proto-1.23.0-py3-none-any.whl.metadata (2.2 kB)\n",
      "Collecting opentelemetry-instrumentation-asgi==0.44b0 (from opentelemetry-instrumentation-fastapi>=0.41b0->chromadb==0.4.21)\n",
      "  Downloading opentelemetry_instrumentation_asgi-0.44b0-py3-none-any.whl.metadata (2.1 kB)\n",
      "Collecting opentelemetry-instrumentation==0.44b0 (from opentelemetry-instrumentation-fastapi>=0.41b0->chromadb==0.4.21)\n",
      "  Downloading opentelemetry_instrumentation-0.44b0-py3-none-any.whl.metadata (6.1 kB)\n",
      "Collecting opentelemetry-semantic-conventions==0.44b0 (from opentelemetry-instrumentation-fastapi>=0.41b0->chromadb==0.4.21)\n",
      "  Downloading opentelemetry_semantic_conventions-0.44b0-py3-none-any.whl.metadata (2.2 kB)\n",
      "Collecting opentelemetry-util-http==0.44b0 (from opentelemetry-instrumentation-fastapi>=0.41b0->chromadb==0.4.21)\n",
      "  Downloading opentelemetry_util_http-0.44b0-py3-none-any.whl.metadata (2.4 kB)\n",
      "Requirement already satisfied: setuptools>=16.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from opentelemetry-instrumentation==0.44b0->opentelemetry-instrumentation-fastapi>=0.41b0->chromadb==0.4.21) (68.2.2)\n",
      "Requirement already satisfied: wrapt<2.0.0,>=1.0.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from opentelemetry-instrumentation==0.44b0->opentelemetry-instrumentation-fastapi>=0.41b0->chromadb==0.4.21) (1.16.0)\n",
      "Collecting asgiref~=3.0 (from opentelemetry-instrumentation-asgi==0.44b0->opentelemetry-instrumentation-fastapi>=0.41b0->chromadb==0.4.21)\n",
      "  Using cached asgiref-3.7.2-py3-none-any.whl.metadata (9.2 kB)\n",
      "Collecting monotonic>=1.5 (from posthog>=2.4.0->chromadb==0.4.21)\n",
      "  Downloading monotonic-1.6-py2.py3-none-any.whl.metadata (1.5 kB)\n",
      "Requirement already satisfied: backoff>=1.10.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from posthog>=2.4.0->chromadb==0.4.21) (2.2.1)\n",
      "Requirement already satisfied: annotated-types>=0.4.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from pydantic>=1.9->chromadb==0.4.21) (0.6.0)\n",
      "Requirement already satisfied: pydantic-core==2.16.3 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from pydantic>=1.9->chromadb==0.4.21) (2.16.3)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests>=2.28->chromadb==0.4.21) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests>=2.28->chromadb==0.4.21) (3.6)\n",
      "Requirement already satisfied: huggingface_hub<1.0,>=0.16.4 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from tokenizers>=0.13.2->chromadb==0.4.21) (0.19.4)\n",
      "Requirement already satisfied: colorama in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from tqdm>=4.65.0->chromadb==0.4.21) (0.4.6)\n",
      "Requirement already satisfied: click<9.0.0,>=7.1.1 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from typer>=0.9.0->chromadb==0.4.21) (8.1.7)\n",
      "Requirement already satisfied: h11>=0.8 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from uvicorn>=0.18.3->uvicorn[standard]>=0.18.3->chromadb==0.4.21) (0.14.0)\n",
      "Collecting httptools>=0.5.0 (from uvicorn[standard]>=0.18.3->chromadb==0.4.21)\n",
      "  Using cached httptools-0.6.1-cp38-cp38-win_amd64.whl.metadata (3.7 kB)\n",
      "Collecting python-dotenv>=0.13 (from uvicorn[standard]>=0.18.3->chromadb==0.4.21)\n",
      "  Downloading python_dotenv-1.0.1-py3-none-any.whl.metadata (23 kB)\n",
      "Collecting watchfiles>=0.13 (from uvicorn[standard]>=0.18.3->chromadb==0.4.21)\n",
      "  Using cached watchfiles-0.21.0-cp38-none-win_amd64.whl.metadata (5.0 kB)\n",
      "Collecting websockets>=10.4 (from uvicorn[standard]>=0.18.3->chromadb==0.4.21)\n",
      "  Using cached websockets-12.0-cp38-cp38-win_amd64.whl.metadata (6.8 kB)\n",
      "Requirement already satisfied: zipp>=3.1.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from importlib-resources->chromadb==0.4.21) (3.18.1)\n",
      "Requirement already satisfied: cachetools<6.0,>=2.0.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from google-auth>=1.0.1->kubernetes>=28.1.0->chromadb==0.4.21) (5.3.3)\n",
      "Collecting pyasn1-modules>=0.2.1 (from google-auth>=1.0.1->kubernetes>=28.1.0->chromadb==0.4.21)\n",
      "  Downloading pyasn1_modules-0.3.0-py2.py3-none-any.whl.metadata (3.6 kB)\n",
      "Collecting rsa<5,>=3.1.4 (from google-auth>=1.0.1->kubernetes>=28.1.0->chromadb==0.4.21)\n",
      "  Downloading rsa-4.9-py3-none-any.whl.metadata (4.2 kB)\n",
      "Requirement already satisfied: filelock in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface_hub<1.0,>=0.16.4->tokenizers>=0.13.2->chromadb==0.4.21) (3.13.1)\n",
      "Requirement already satisfied: fsspec>=2023.5.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface_hub<1.0,>=0.16.4->tokenizers>=0.13.2->chromadb==0.4.21) (2024.3.0)\n",
      "Requirement already satisfied: anyio<5,>=3.4.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from starlette<0.37.0,>=0.36.3->fastapi>=0.95.2->chromadb==0.4.21) (4.3.0)\n",
      "Collecting humanfriendly>=9.1 (from coloredlogs->onnxruntime>=1.14.1->chromadb==0.4.21)\n",
      "  Downloading humanfriendly-10.0-py2.py3-none-any.whl.metadata (9.2 kB)\n",
      "Requirement already satisfied: mpmath>=0.19 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sympy->onnxruntime>=1.14.1->chromadb==0.4.21) (1.3.0)\n",
      "Requirement already satisfied: sniffio>=1.1 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from anyio<5,>=3.4.0->starlette<0.37.0,>=0.36.3->fastapi>=0.95.2->chromadb==0.4.21) (1.3.1)\n",
      "Requirement already satisfied: exceptiongroup>=1.0.2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from anyio<5,>=3.4.0->starlette<0.37.0,>=0.36.3->fastapi>=0.95.2->chromadb==0.4.21) (1.2.0)\n",
      "Collecting pyreadline3 (from humanfriendly>=9.1->coloredlogs->onnxruntime>=1.14.1->chromadb==0.4.21)\n",
      "  Downloading pyreadline3-3.4.1-py3-none-any.whl.metadata (2.0 kB)\n",
      "Collecting pyasn1<0.6.0,>=0.4.6 (from pyasn1-modules>=0.2.1->google-auth>=1.0.1->kubernetes>=28.1.0->chromadb==0.4.21)\n",
      "  Using cached pyasn1-0.5.1-py2.py3-none-any.whl.metadata (8.6 kB)\n",
      "Using cached chromadb-0.4.21-py3-none-any.whl (508 kB)\n",
      "Using cached chroma_hnswlib-0.7.3-cp38-cp38-win_amd64.whl (150 kB)\n",
      "Using cached bcrypt-4.1.2-cp37-abi3-win_amd64.whl (158 kB)\n",
      "Downloading fastapi-0.110.0-py3-none-any.whl (92 kB)\n",
      "   ---------------------------------------- 0.0/92.1 kB ? eta -:--:--\n",
      "   ---------------------------------------- 92.1/92.1 kB 5.5 MB/s eta 0:00:00\n",
      "Downloading graphlib_backport-1.1.0-py3-none-any.whl (7.1 kB)\n",
      "Downloading grpcio-1.62.1-cp38-cp38-win_amd64.whl (3.8 MB)\n",
      "   ---------------------------------------- 0.0/3.8 MB ? eta -:--:--\n",
      "   ------- -------------------------------- 0.7/3.8 MB 15.3 MB/s eta 0:00:01\n",
      "   --------------- ------------------------ 1.5/3.8 MB 13.2 MB/s eta 0:00:01\n",
      "   --------------------- ------------------ 2.0/3.8 MB 12.8 MB/s eta 0:00:01\n",
      "   -------------------------- ------------- 2.5/3.8 MB 12.5 MB/s eta 0:00:01\n",
      "   -------------------------------- ------- 3.1/3.8 MB 12.4 MB/s eta 0:00:01\n",
      "   -------------------------------------- - 3.7/3.8 MB 12.3 MB/s eta 0:00:01\n",
      "   ---------------------------------------- 3.8/3.8 MB 11.6 MB/s eta 0:00:00\n",
      "Downloading kubernetes-29.0.0-py2.py3-none-any.whl (1.6 MB)\n",
      "   ---------------------------------------- 0.0/1.6 MB ? eta -:--:--\n",
      "   ---------------- ----------------------- 0.6/1.6 MB 13.5 MB/s eta 0:00:01\n",
      "   ------------------------------ --------- 1.2/1.6 MB 12.6 MB/s eta 0:00:01\n",
      "   ---------------------------------------  1.6/1.6 MB 12.6 MB/s eta 0:00:01\n",
      "   ---------------------------------------- 1.6/1.6 MB 11.2 MB/s eta 0:00:00\n",
      "Downloading mmh3-4.1.0-cp38-cp38-win_amd64.whl (31 kB)\n",
      "Downloading onnxruntime-1.17.1-cp38-cp38-win_amd64.whl (5.6 MB)\n",
      "   ---------------------------------------- 0.0/5.6 MB ? eta -:--:--\n",
      "   ---- ----------------------------------- 0.6/5.6 MB 13.1 MB/s eta 0:00:01\n",
      "   ------- -------------------------------- 1.0/5.6 MB 10.9 MB/s eta 0:00:01\n",
      "   ---------- ----------------------------- 1.5/5.6 MB 10.8 MB/s eta 0:00:01\n",
      "   -------------- ------------------------- 2.1/5.6 MB 10.9 MB/s eta 0:00:01\n",
      "   ------------------ --------------------- 2.6/5.6 MB 11.1 MB/s eta 0:00:01\n",
      "   ---------------------- ----------------- 3.1/5.6 MB 11.1 MB/s eta 0:00:01\n",
      "   ------------------------- -------------- 3.6/5.6 MB 11.5 MB/s eta 0:00:01\n",
      "   ----------------------------- ---------- 4.1/5.6 MB 11.3 MB/s eta 0:00:01\n",
      "   ---------------------------------- ----- 4.9/5.6 MB 11.5 MB/s eta 0:00:01\n",
      "   ---------------------------------------  5.6/5.6 MB 11.5 MB/s eta 0:00:01\n",
      "   ---------------------------------------- 5.6/5.6 MB 10.5 MB/s eta 0:00:00\n",
      "Downloading opentelemetry_api-1.23.0-py3-none-any.whl (58 kB)\n",
      "   ---------------------------------------- 0.0/58.4 kB ? eta -:--:--\n",
      "   ---------------------------------------- 58.4/58.4 kB 3.0 MB/s eta 0:00:00\n",
      "Downloading opentelemetry_exporter_otlp_proto_grpc-1.23.0-py3-none-any.whl (18 kB)\n",
      "Downloading opentelemetry_exporter_otlp_proto_common-1.23.0-py3-none-any.whl (17 kB)\n",
      "Downloading opentelemetry_proto-1.23.0-py3-none-any.whl (50 kB)\n",
      "   ---------------------------------------- 0.0/50.8 kB ? eta -:--:--\n",
      "   ---------------------------------------- 50.8/50.8 kB 2.7 MB/s eta 0:00:00\n",
      "Downloading opentelemetry_instrumentation_fastapi-0.44b0-py3-none-any.whl (11 kB)\n",
      "Downloading opentelemetry_instrumentation-0.44b0-py3-none-any.whl (28 kB)\n",
      "Downloading opentelemetry_instrumentation_asgi-0.44b0-py3-none-any.whl (14 kB)\n",
      "Downloading opentelemetry_semantic_conventions-0.44b0-py3-none-any.whl (36 kB)\n",
      "Downloading opentelemetry_util_http-0.44b0-py3-none-any.whl (6.9 kB)\n",
      "Downloading opentelemetry_sdk-1.23.0-py3-none-any.whl (105 kB)\n",
      "   ---------------------------------------- 0.0/105.7 kB ? eta -:--:--\n",
      "   ---------------------------------------- 105.7/105.7 kB 6.0 MB/s eta 0:00:00\n",
      "Downloading overrides-7.7.0-py3-none-any.whl (17 kB)\n",
      "Downloading posthog-3.5.0-py2.py3-none-any.whl (41 kB)\n",
      "   ---------------------------------------- 0.0/41.3 kB ? eta -:--:--\n",
      "   ---------------------------------------- 41.3/41.3 kB 1.9 MB/s eta 0:00:00\n",
      "Downloading pulsar_client-3.4.0-cp38-cp38-win_amd64.whl (3.4 MB)\n",
      "   ---------------------------------------- 0.0/3.4 MB ? eta -:--:--\n",
      "   ------- -------------------------------- 0.7/3.4 MB 14.2 MB/s eta 0:00:01\n",
      "   -------------- ------------------------- 1.2/3.4 MB 12.9 MB/s eta 0:00:01\n",
      "   -------------------- ------------------- 1.8/3.4 MB 12.5 MB/s eta 0:00:01\n",
      "   --------------------------- ------------ 2.3/3.4 MB 12.4 MB/s eta 0:00:01\n",
      "   --------------------------------- ------ 2.9/3.4 MB 13.0 MB/s eta 0:00:01\n",
      "   ---------------------------------------  3.4/3.4 MB 12.1 MB/s eta 0:00:01\n",
      "   ---------------------------------------- 3.4/3.4 MB 11.0 MB/s eta 0:00:00\n",
      "Using cached tokenizers-0.15.2-cp38-none-win_amd64.whl (2.2 MB)\n",
      "Using cached typer-0.9.0-py3-none-any.whl (45 kB)\n",
      "Downloading uvicorn-0.28.0-py3-none-any.whl (60 kB)\n",
      "   ---------------------------------------- 0.0/60.6 kB ? eta -:--:--\n",
      "   ---------------------------------------- 60.6/60.6 kB 3.1 MB/s eta 0:00:00\n",
      "Using cached Deprecated-1.2.14-py2.py3-none-any.whl (9.6 kB)\n",
      "Downloading google_auth-2.28.2-py2.py3-none-any.whl (186 kB)\n",
      "   ---------------------------------------- 0.0/186.9 kB ? eta -:--:--\n",
      "   ---------------------------------------- 186.9/186.9 kB 5.7 MB/s eta 0:00:00\n",
      "Downloading googleapis_common_protos-1.63.0-py2.py3-none-any.whl (229 kB)\n",
      "   ---------------------------------------- 0.0/229.1 kB ? eta -:--:--\n",
      "   --------------------------------------- 229.1/229.1 kB 13.7 MB/s eta 0:00:00\n",
      "Using cached httptools-0.6.1-cp38-cp38-win_amd64.whl (60 kB)\n",
      "Using cached monotonic-1.6-py2.py3-none-any.whl (8.2 kB)\n",
      "Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)\n",
      "Downloading python_dotenv-1.0.1-py3-none-any.whl (19 kB)\n",
      "Downloading starlette-0.36.3-py3-none-any.whl (71 kB)\n",
      "   ---------------------------------------- 0.0/71.5 kB ? eta -:--:--\n",
      "   ---------------------------------------- 71.5/71.5 kB 3.8 MB/s eta 0:00:00\n",
      "Using cached watchfiles-0.21.0-cp38-none-win_amd64.whl (279 kB)\n",
      "Using cached websocket_client-1.7.0-py3-none-any.whl (58 kB)\n",
      "Using cached websockets-12.0-cp38-cp38-win_amd64.whl (124 kB)\n",
      "Using cached coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n",
      "Downloading flatbuffers-24.3.7-py2.py3-none-any.whl (26 kB)\n",
      "Downloading requests_oauthlib-1.4.0-py2.py3-none-any.whl (24 kB)\n",
      "Using cached asgiref-3.7.2-py3-none-any.whl (24 kB)\n",
      "Using cached humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n",
      "Using cached pyasn1_modules-0.3.0-py2.py3-none-any.whl (181 kB)\n",
      "Using cached rsa-4.9-py3-none-any.whl (34 kB)\n",
      "Using cached pyasn1-0.5.1-py2.py3-none-any.whl (84 kB)\n",
      "Using cached pyreadline3-3.4.1-py3-none-any.whl (95 kB)\n",
      "Installing collected packages: pyreadline3, pypika, monotonic, mmh3, flatbuffers, websockets, websocket-client, python-dotenv, pyasn1, pulsar-client, overrides, opentelemetry-util-http, opentelemetry-semantic-conventions, opentelemetry-proto, oauthlib, humanfriendly, httptools, grpcio, graphlib-backport, googleapis-common-protos, deprecated, chroma-hnswlib, bcrypt, asgiref, watchfiles, uvicorn, typer, starlette, rsa, requests-oauthlib, pyasn1-modules, posthog, opentelemetry-exporter-otlp-proto-common, opentelemetry-api, coloredlogs, tokenizers, opentelemetry-sdk, opentelemetry-instrumentation, onnxruntime, google-auth, fastapi, opentelemetry-instrumentation-asgi, opentelemetry-exporter-otlp-proto-grpc, kubernetes, opentelemetry-instrumentation-fastapi, chromadb\n",
      "  Attempting uninstall: tokenizers\n",
      "    Found existing installation: tokenizers 0.10.3\n",
      "    Uninstalling tokenizers-0.10.3:\n",
      "      Successfully uninstalled tokenizers-0.10.3\n",
      "Successfully installed asgiref-3.7.2 bcrypt-4.1.2 chroma-hnswlib-0.7.3 chromadb-0.4.21 coloredlogs-15.0.1 deprecated-1.2.14 fastapi-0.110.0 flatbuffers-24.3.7 google-auth-2.28.2 googleapis-common-protos-1.63.0 graphlib-backport-1.1.0 grpcio-1.62.1 httptools-0.6.1 humanfriendly-10.0 kubernetes-29.0.0 mmh3-4.1.0 monotonic-1.6 oauthlib-3.2.2 onnxruntime-1.17.1 opentelemetry-api-1.23.0 opentelemetry-exporter-otlp-proto-common-1.23.0 opentelemetry-exporter-otlp-proto-grpc-1.23.0 opentelemetry-instrumentation-0.44b0 opentelemetry-instrumentation-asgi-0.44b0 opentelemetry-instrumentation-fastapi-0.44b0 opentelemetry-proto-1.23.0 opentelemetry-sdk-1.23.0 opentelemetry-semantic-conventions-0.44b0 opentelemetry-util-http-0.44b0 overrides-7.7.0 posthog-3.5.0 pulsar-client-3.4.0 pyasn1-0.5.1 pyasn1-modules-0.3.0 pypika-0.48.9 pyreadline3-3.4.1 python-dotenv-1.0.1 requests-oauthlib-1.4.0 rsa-4.9 starlette-0.36.3 tokenizers-0.15.2 typer-0.9.0 uvicorn-0.28.0 watchfiles-0.21.0 websocket-client-1.7.0 websockets-12.0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "    WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "  WARNING: Failed to remove contents in a temporary directory 'E:\\anaconda3\\envs\\error_env\\Lib\\site-packages\\~okenizers'.\n",
      "  You can safely remove it manually.\n",
      "ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "transformers 4.12.5 requires tokenizers<0.11,>=0.10.1, but you have tokenizers 0.15.2 which is incompatible.\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n"
     ]
    }
   ],
   "source": [
    "!pip install chromadb==0.4.21"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e55467e4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting openai==1.4.0\n",
      "  Using cached openai-1.4.0-py3-none-any.whl.metadata (17 kB)\n",
      "Requirement already satisfied: anyio<5,>=3.5.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from openai==1.4.0) (4.3.0)\n",
      "Requirement already satisfied: distro<2,>=1.7.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from openai==1.4.0) (1.9.0)\n",
      "Requirement already satisfied: httpx<1,>=0.23.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from openai==1.4.0) (0.27.0)\n",
      "Requirement already satisfied: pydantic<3,>=1.9.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from openai==1.4.0) (2.6.4)\n",
      "Requirement already satisfied: sniffio in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from openai==1.4.0) (1.3.1)\n",
      "Requirement already satisfied: tqdm>4 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from openai==1.4.0) (4.66.2)\n",
      "Requirement already satisfied: typing-extensions<5,>=4.5 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from openai==1.4.0) (4.10.0)\n",
      "Requirement already satisfied: idna>=2.8 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from anyio<5,>=3.5.0->openai==1.4.0) (3.6)\n",
      "Requirement already satisfied: exceptiongroup>=1.0.2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from anyio<5,>=3.5.0->openai==1.4.0) (1.2.0)\n",
      "Requirement already satisfied: certifi in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from httpx<1,>=0.23.0->openai==1.4.0) (2024.2.2)\n",
      "Requirement already satisfied: httpcore==1.* in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from httpx<1,>=0.23.0->openai==1.4.0) (1.0.4)\n",
      "Requirement already satisfied: h11<0.15,>=0.13 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from httpcore==1.*->httpx<1,>=0.23.0->openai==1.4.0) (0.14.0)\n",
      "Requirement already satisfied: annotated-types>=0.4.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from pydantic<3,>=1.9.0->openai==1.4.0) (0.6.0)\n",
      "Requirement already satisfied: pydantic-core==2.16.3 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from pydantic<3,>=1.9.0->openai==1.4.0) (2.16.3)\n",
      "Requirement already satisfied: colorama in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from tqdm>4->openai==1.4.0) (0.4.6)\n",
      "Using cached openai-1.4.0-py3-none-any.whl (221 kB)\n",
      "Installing collected packages: openai\n",
      "  Attempting uninstall: openai\n",
      "    Found existing installation: openai 0.28.1\n",
      "    Uninstalling openai-0.28.1:\n",
      "      Successfully uninstalled openai-0.28.1\n",
      "Successfully installed openai-1.4.0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "    WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n"
     ]
    }
   ],
   "source": [
    "!pip install openai==1.4.0 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "cbb7e080",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: sentence-transformers==2.2.2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (2.2.2)\n",
      "Requirement already satisfied: transformers<5.0.0,>=4.6.0 in c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages (from sentence-transformers==2.2.2) (4.12.5)\n",
      "Requirement already satisfied: tqdm in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (4.66.2)\n",
      "Requirement already satisfied: torch>=1.6.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (2.2.1)\n",
      "Requirement already satisfied: torchvision in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (0.17.1)\n",
      "Requirement already satisfied: numpy in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (1.24.4)\n",
      "Requirement already satisfied: scikit-learn in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (1.3.2)\n",
      "Requirement already satisfied: scipy in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (1.10.1)\n",
      "Requirement already satisfied: nltk in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (3.8.1)\n",
      "Requirement already satisfied: sentencepiece in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (0.2.0)\n",
      "Requirement already satisfied: huggingface-hub>=0.4.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sentence-transformers==2.2.2) (0.19.4)\n",
      "Requirement already satisfied: filelock in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (3.13.1)\n",
      "Requirement already satisfied: fsspec>=2023.5.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (2024.3.0)\n",
      "Requirement already satisfied: requests in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (2.31.0)\n",
      "Requirement already satisfied: pyyaml>=5.1 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (6.0.1)\n",
      "Requirement already satisfied: typing-extensions>=3.7.4.3 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (4.10.0)\n",
      "Requirement already satisfied: packaging>=20.9 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (23.2)\n",
      "Requirement already satisfied: sympy in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from torch>=1.6.0->sentence-transformers==2.2.2) (1.12)\n",
      "Requirement already satisfied: networkx in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from torch>=1.6.0->sentence-transformers==2.2.2) (3.1)\n",
      "Requirement already satisfied: jinja2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from torch>=1.6.0->sentence-transformers==2.2.2) (3.1.3)\n",
      "Requirement already satisfied: colorama in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from tqdm->sentence-transformers==2.2.2) (0.4.6)\n",
      "Requirement already satisfied: regex!=2019.12.17 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from transformers<5.0.0,>=4.6.0->sentence-transformers==2.2.2) (2023.12.25)\n",
      "Requirement already satisfied: sacremoses in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from transformers<5.0.0,>=4.6.0->sentence-transformers==2.2.2) (0.1.1)\n",
      "Collecting tokenizers<0.11,>=0.10.1 (from transformers<5.0.0,>=4.6.0->sentence-transformers==2.2.2)\n",
      "  Using cached tokenizers-0.10.3-cp38-cp38-win_amd64.whl.metadata (5.8 kB)\n",
      "Requirement already satisfied: click in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from nltk->sentence-transformers==2.2.2) (8.1.7)\n",
      "Requirement already satisfied: joblib in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from nltk->sentence-transformers==2.2.2) (1.3.2)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from scikit-learn->sentence-transformers==2.2.2) (3.3.0)\n",
      "Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from torchvision->sentence-transformers==2.2.2) (10.2.0)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from jinja2->torch>=1.6.0->sentence-transformers==2.2.2) (2.1.5)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (3.6)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (2.2.1)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from requests->huggingface-hub>=0.4.0->sentence-transformers==2.2.2) (2024.2.2)\n",
      "Requirement already satisfied: mpmath>=0.19 in e:\\anaconda3\\envs\\error_env\\lib\\site-packages (from sympy->torch>=1.6.0->sentence-transformers==2.2.2) (1.3.0)\n",
      "Using cached tokenizers-0.10.3-cp38-cp38-win_amd64.whl (2.0 MB)\n",
      "Installing collected packages: tokenizers\n",
      "  Attempting uninstall: tokenizers\n",
      "    Found existing installation: tokenizers 0.15.2\n",
      "    Uninstalling tokenizers-0.15.2:\n",
      "      Successfully uninstalled tokenizers-0.15.2\n",
      "Successfully installed tokenizers-0.10.3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "    WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "chromadb 0.4.21 requires tokenizers>=0.13.2, but you have tokenizers 0.10.3 which is incompatible.\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -pype1 (c:\\users\\jyseo\\appdata\\roaming\\python\\python38\\site-packages)\n"
     ]
    }
   ],
   "source": [
    "!pip install sentence-transformers==2.2.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1be69cf2",
   "metadata": {},
   "outputs": [],
   "source": [
    "from langchain.document_loaders import TextLoader\n",
    "documents = TextLoader(\"e:/data/AI.txt\").load()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "124a3fe4",
   "metadata": {},
   "outputs": [],
   "source": [
    "from langchain.text_splitter import RecursiveCharacterTextSplitter\n",
    "\n",
    "# 문서를 청크로 분할\n",
    "def split_docs(documents,chunk_size=1000,chunk_overlap=20):\n",
    "  text_splitter = RecursiveCharacterTextSplitter(chunk_size=chunk_size, chunk_overlap=chunk_overlap)\n",
    "  docs = text_splitter.split_documents(documents)\n",
    "  return docs\n",
    "\n",
    "# docs 변수에 분할 문서를 저장\n",
    "docs = split_docs(documents)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6434e36c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "e:\\anaconda3\\envs\\error_env\\lib\\site-packages\\tqdm\\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "from langchain.embeddings import SentenceTransformerEmbeddings\n",
    "embeddings = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
    "\n",
    "# Chromdb에 벡터 저장\n",
    "from langchain.vectorstores import Chroma\n",
    "db = Chroma.from_documents(docs, embeddings)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "8ea8ff78",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "e:\\anaconda3\\envs\\error_env\\lib\\site-packages\\langchain_core\\_api\\deprecation.py:117: LangChainDeprecationWarning: The class `langchain_community.chat_models.openai.ChatOpenAI` was deprecated in langchain-community 0.0.10 and will be removed in 0.2.0. An updated version of the class exists in the langchain-openai package and should be used instead. To use it run `pip install -U langchain-openai` and import as `from langchain_openai import ChatOpenAI`.\n",
      "  warn_deprecated(\n",
      "Number of requested results 4 is greater than number of elements in index 3, updating n_results = 3\n",
      "e:\\anaconda3\\envs\\error_env\\lib\\site-packages\\langchain_core\\_api\\deprecation.py:117: LangChainDeprecationWarning: The function `run` was deprecated in LangChain 0.1.0 and will be removed in 0.2.0. Use invoke instead.\n",
      "  warn_deprecated(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\u001b[1m> Entering new StuffDocumentsChain chain...\u001b[0m\n",
      "\n",
      "\n",
      "\u001b[1m> Entering new LLMChain chain...\u001b[0m\n",
      "Prompt after formatting:\n",
      "\u001b[32;1m\u001b[1;3mSystem: Use the following pieces of context to answer the user's question. \n",
      "If you don't know the answer, just say that you don't know, don't try to make up an answer.\n",
      "----------------\n",
      "Artificial intelligence (AI) is the intelligence of machines or software, as opposed to the intelligence of humans or animals. It is a field of study in computer science that develops and studies intelligent machines. Such machines may be called AIs.\n",
      "\n",
      "AI technology is widely used throughout industry, government, and science. Some high-profile applications are: advanced web search engines (e.g., Google Search), recommendation systems (used by YouTube, Amazon, and Netflix), understanding human speech (such as Google Assistant, Siri, and Alexa), self-driving cars (e.g., Waymo), generative and creative tools (ChatGPT and AI art), and superhuman play and analysis in strategy games (such as chess and Go).[1]\n",
      "\n",
      "The various sub-fields of AI research are centered around particular goals and the use of particular tools. The traditional goals of AI research include reasoning, knowledge representation, planning, learning, natural language processing, perception, and support for robotics.[a] General intelligence (the ability to complete any task performable by a human) is among the field's long-term goals.[11]\n",
      "\n",
      "To solve these problems, AI researchers have adapted and integrated a wide range of problem-solving techniques, including search and mathematical optimization, formal logic, artificial neural networks, and methods based on statistics, operations research, and economics.[b] AI also draws upon psychology, linguistics, philosophy, neuroscience and other fields.[12]\n",
      "\n",
      "Alan Turing was the first person to carry out substantial research in the field that he called Machine Intelligence.[2] Artificial intelligence was founded as an academic discipline in 1956.[3] The field went through multiple cycles of optimism[4][5] followed by disappointment and loss of funding.[6][7] Funding and interest vastly increased after 2012 when deep learning surpassed all previous AI techniques,[8] and after 2017 with the transformer architecture.[9] This led to the AI spring of the 2020s, with companies, universities, and laboratories overwhelmingly based in the United States pioneering significant advances in artificial intelligence.[10]\n",
      "Human: AI란?\u001b[0m\n",
      "\n",
      "\u001b[1m> Finished chain.\u001b[0m\n",
      "\n",
      "\u001b[1m> Finished chain.\u001b[0m\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'AI는 기계나 소프트웨어의 지능을 의미하며, 사람이나 동물의 지능과 구분됩니다. 컴퓨터 과학의 한 분야로, 지능 있는 기계를 개발하고 연구하는 분야입니다. AI 기술은 산업, 정부, 과학 분야에서 널리 사용되며 다양한 응용 분야가 있습니다. AI 연구의 다양한 하위 분야는 특정 목표와 도구의 사용을 중심으로 하고 있습니다.'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os #운영체제(os) 모듈을 가져올 때 사용되는 라이브러리\n",
    "os.environ[\"OPENAI_API_KEY\"] = \"sk-\" #openai 키 입력\n",
    "\n",
    "from langchain.chat_models import ChatOpenAI\n",
    "model_name = \"gpt-3.5-turbo\"  #GPT-3.5 turbo 모델 사용\n",
    "llm = ChatOpenAI(model_name=model_name)\n",
    "\n",
    "# Q&A 체인을 사용하여 쿼리에 대한 답변 얻기\n",
    "from langchain.chains.question_answering import load_qa_chain\n",
    "chain = load_qa_chain(llm, chain_type=\"stuff\",verbose=True)\n",
    "\n",
    "# 쿼리를 작성하고 유사성 검색을 수행하여 답변을 생성,따라서 txt에 있는 내용을 질의해야 합니다\n",
    "query = \"AI란?\"\n",
    "matching_docs = db.similarity_search(query)\n",
    "answer =  chain.run(input_documents=matching_docs, question=query)\n",
    "answer\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27a686a4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "error_env",
   "language": "python",
   "name": "error_env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
